sdp/processors/create_initial_manifest/create_initial_manifest_voxpopuli.py

```diff
@@ -0,0 +1,145 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+from pathlib import Path
+
+import sox
+from nemo.utils import logging
+from sox import Transformer
+
+from sdp.processors.base_processor import BaseParallelProcessor, DataEntry
+from sdp.utils.common import extract_archive
+
+VOXPOPULI_URL = "https://github.com/facebookresearch/voxpopuli"
+
+
+class CreateInitialManifestVoxpopuli(BaseParallelProcessor):
+    """
+    Downloads and unzips raw VoxPopuli data for the specified language,
+    and creates an initial manifest using the transcripts provided in the
+    raw data.
+
+    Args:
+        language_id: the language of the data you wish to be downloaded.
+        download_dir: the directory where the downloaded data will be saved.
+        data_split: the data split for which the initial manifest will
+            be created.
+        resampled_audio_dir: the directory where the resampled (16kHz) wav
+            files will be stored.
+        use_test_data: if `True`, will use the test data manifest located
+            at `TEST_DATA_PATH` to carry out tests.
+    """
+
+    def __init__(
+        self,
+        language_id: str,
+        download_dir: str,
+        resampled_audio_dir: str,
+        data_split: str,
+        use_test_data: bool = False,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.language_id = language_id
+        self.download_dir = Path(download_dir)
+        self.data_split = data_split
+        self.resampled_audio_dir = resampled_audio_dir
+        self.use_test_data = use_test_data
+
+    def prepare(self):
+        """Downloading and extracting data (unless already done).
+
+        If use_test_data is True, then will not download data, instead will
+        copy the included test data (mainly useful for quick development or
+        CI pipeline).
+        """
+        if self.use_test_data:
+            try:
+                __TEST_DATA_ROOT = os.environ["TEST_DATA_ROOT"]
+                logging.info(f"Found 'TEST_DATA_ROOT' environment variable:{repr(__TEST_DATA_ROOT)}")
+            except KeyError:
+                raise KeyError(
+                    f"Tried to look for os.environ['TEST_DATA_ROOT'] but it was not set."
+                    f" Please set 'TEST_DATA_ROOT' as an environment variable and try again."
+                )
+
+            self.test_data_path = str(Path(__TEST_DATA_ROOT) / self.language / "voxpopuli" / "data.tar.gz")
+
+            if not os.path.exists(self.test_data_path):
+                raise ValueError(
+                    f"No such file {self.test_data_path}. Are you sure you specified the "
+                    f" 'TEST_DATA_ROOT' environment variable correctly?"
+                )
+            data_folder = extract_archive(str(self.test_data_path), str(self.download_dir))
+        # else:
+        # TODO: some kind of isolated environment?
```

Comment:
Yes, it isn't ideal that we modify the current environment by installing things but I don't know of a good workaround.  I think it's fine for now.